### ISSUE TITLE
`Layout/SpaceBeforeBrackets`: false positive for arguments forwarding to `Hash#[]`

### ISSUE THREAD

**Issue by yujideveloper** [rubocop/rubocop#10573]
## Expected behavior

It does not detect offense of `Layout/SpaceBeforeBrackets`.

## Actual behavior

```
$ rubocop --debug --cache false --only Layout/SpaceBeforeBrackets foo.rb
For /Users/yuji_developer/foo: configuration from /Users/yuji_developer/foo/.rubocop.yml
Default configuration from /Users/yuji_developer/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/rubocop-1.28.1/config/default.yml
Inspecting 1 file
Scanning /Users/yuji_developer/foo/foo.rb
C

Offenses:

foo.rb:7:10: C: [Correctable] Layout/SpaceBeforeBrackets: Remove the space before the opening brackets.
    @hash.[](...)
         ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.06047500018030405 seconds
```

## Steps to reproduce the problem

Run `rubocop --only Layout/SpaceBeforeBrackets foo.rb` on the following config and code.

```yaml
# .rubocop.yml
AllCops:
  TargetRubyVersion: 2.7
  NewCops: disable
```
```ruby
# foo.rb
class Foo
  def initialize
    @hash = {}
  end

  def [](...)
    @hash.[](...)
  end
end
```

## RuboCop version

```
$ rubocop -V
1.28.1 (using Parser 3.1.2.0, rubocop-ast 1.17.0, running on ruby 2.7.6 arm64-darwin21)
```
